app/controllers/questionnaire_controller.rb

```diff
@@ -211,6 +213,20 @@ def check_content_hash
     end
   end
 
+  def check_content_empty
+    quest_content = questionnaire_content
+    return if quest_content.present? && (quest_content.keys - [Response::CSRF_FAILED]).present?
+
+    render(status: :bad_request, html: 'Cannot store blank questionnaire responses', layout: 'application')
+  end
+
+  def check_opened_at
+    return if @response.opened_at.present?
+
+    render(status: :bad_request, html: 'Cannot accept answers for an unopened response',
```

Comment:
hehe nice.

Reply:
I still have no clue how they did this in the first place. The only way I see possibled is if they used the API to read out the list of questionnaires.

Reply:
Either way, this fixes it.